core/src/utils/focus-controller/test/ionic/focus-controller.e2e.ts

```diff
@@ -0,0 +1,63 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+configs({ modes: ['md'], directions: ['ltr'] }).forEach(({ title, config }) => {
+  test.describe(title('focus controller: ionic components'), () => {
+    test.beforeEach(async ({ page }) => {
+      await page.goto('/src/utils/focus-controller/test/ionic', config);
+    });
+    test('should focus heading', async ({ page }) => {
+      const goToPageOneButton = page.locator('page-root ion-button.page-one');
+      const nav = page.locator('ion-nav');
+      const ionNavDidChange = await (nav as any).spyOnEvent('ionNavDidChange');
```

Comment:
I need to figure out why `spyOnEvent` does not exist on the type interface for our custom Locator, but the method does indeed exist.

Reply:
You need to cast:
```ts
import type { E2ELocator } from '@utils/test/playwright';
...
const nav = page.locator('ion-nav') as E2ELocator;
```